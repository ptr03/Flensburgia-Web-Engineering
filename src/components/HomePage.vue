<!-- src/pages/HomePage.vue -->
<template>
  <HeroSection />

  <!-- About Teaser -->
  <section class="section about-teaser">
    <div class="container">
      <h2 class="section-title text-center animate-on-scroll">Über Flensburgia</h2>
      <p class="teaser-text animate-on-scroll">
        Die Schlaraffia Flensburgia ist Teil der weltweiten Vereinigung Schlaraffia® – einer Gemeinschaft zur Pflege von Freundschaft, Kunst und Humor.
      </p>
      <div class="stats-grid">
        <div
          v-for="(stat, index) in statsTeaser"
          :key="index"
          class="stat-item animate-on-scroll"
          :style="{ 'transition-delay': `${0.2 * index}s` }"
        >
          <component :is="stat.icon" size="32" class="stat-icon text-primary" />
          <h3 class="stat-title">{{ stat.title }}</h3>
          <p class="stat-value">{{ stat.value }}</p>
        </div>
      </div>
      <div class="cta-wrapper text-center">
        <router-link to="/about" class="btn btn-outline-gradient mt-4 animate-on-scroll">
          Mehr über uns
        </router-link>
      </div>
    </div>
  </section>

  <!-- Aktuelles Section (in Blöcken, Hintergrund: weiß) -->
  <section class="section aktuelles-teaser">
    <div class="container">
      <h2 class="section-title text-center animate-on-scroll">Aktuelles</h2>
      <div class="aktuales-grid">
        <div
          v-for="(item, idx) in aktuales"
          :key="item.id"
          class="aktual-card animate-on-scroll"
          :style="{ 'transition-delay': `${(idx + 1) * 0.2}s` }"
        >
          <p class="aktual-text">{{ item.text }}</p>
        </div>
      </div>
    </div>
  </section>

  <!-- Events Teaser (Kommende Veranstaltungen, verwaltet per JSON, Hintergrund: weiß) -->
  <section class="section events-teaser">
    <div class="container">
      <h2 class="section-title text-center animate-on-scroll">Kommende Veranstaltungen</h2>
      <div class="events-grid">
        <article
          v-for="(event, idx) in upcomingEvents"
          :key="event.id"
          class="event-card animate-on-scroll"
          :style="{ 'transition-delay': `${(idx + 1) * 0.2}s` }"
        >
          <time class="event-date">{{ event.date }}</time>
          <h3 class="event-title">{{ event.title }}</h3>
          <p class="event-excerpt">{{ event.excerpt }}</p>
        </article>
      </div>
      <div class="cta-wrapper text-center">
        <router-link to="/events" class="btn btn-outline-gradient mt-4 animate-on-scroll">
          Alle Veranstaltungen
        </router-link>
      </div>
    </div>
  </section>
</template>

<script setup>
import { onMounted, ref } from 'vue'
import HeroSection from '../components/HeroSection.vue'
import { CalendarDays, Trophy, Users } from 'lucide-vue-next'
import aktualesData from '../data/aktuelles.json'
import eventsData from '../data/events.json'

// Statistiken für das "Über Flensburgia"
const statsTeaser = ref([
  { title: 'Jährliche Veranstaltungen', value: '30+', icon: CalendarDays },
  { title: 'Gegründet', value: '1973', icon: Trophy },
  { title: 'Aktive Mitglieder', value: '25+', icon: Users }
])

// "Aktuelles" aus JSON-Datei
const aktuales = ref(aktualesData)

// "Kommende Veranstaltungen" aus JSON-Datei
const allEvents = ref(eventsData)
const upcomingEvents = ref(allEvents.value.slice(0, 3))

// IntersectionObserver für Scroll-Animation
onMounted(() => {
  const observer = new IntersectionObserver(
    entries => {
      entries.forEach(entry => {
        if (entry.isIntersecting) {
          entry.target.classList.add('animate-in')
        }
      })
    },
    { threshold: 0.1 }
  )
  document.querySelectorAll('.animate-on-scroll').forEach(el => observer.observe(el))
})
</script>

<style scoped>
.section {
  padding: clamp(4rem, 8vw, 6rem) 1.5rem;
}

/* About Teaser */
.about-teaser {
  background-color: #ffffff; /* weißer Block */
}
.about-teaser .section-title {
  color: #2b2e32;
  font-size: clamp(1.875rem, 4vw, 2.5rem);
  font-weight: 600;
  margin-bottom: 1rem;
}
.about-teaser .teaser-text {
  color: #2f3034;
  margin-bottom: 2rem;
  font-size: clamp(1rem, 1.2vw, 1.125rem);
  line-height: 1.6;
}
.stats-grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  gap: 2rem;
  margin-bottom: 3rem;
}
.stat-item {
  background-color: #ffffff;
  border-radius: 16px;
  padding: 2rem;
  text-align: center;
  box-shadow: 0 4px 16px rgba(31, 41, 55, 0.1);
}
.stat-icon { color: #2563eb; margin-bottom: 0.75rem; }
.stat-title { font-size: 1rem; font-weight: 500; margin-bottom: 0.25rem; color: #111827; }
.stat-value { font-size: 1.75rem; font-weight: 700; color: #111827; }
.cta-wrapper { display: flex; justify-content: center; margin-top: 2rem; }

/* Aktuelles Teaser (weißer Block) */
.aktuelles-teaser {
  background-color: #ffffff;
}
.aktuelles-teaser .section-title {
  color: #111827;
  font-size: clamp(1.875rem, 4vw, 2.5rem);
  font-weight: 600;
  margin-bottom: 1.5rem;
}
.aktuales-grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(280px, 1fr));
  gap: 2rem;
}
.aktual-card {
  background-color: #ffffff;
  border-radius: 16px;
  padding: 1.5rem;
  box-shadow: 0 4px 16px rgba(31, 41, 55, 0.05);
  transition: transform 0.3s ease;
}
.aktual-card:hover {
  transform: translateY(-4px);
}
.aktual-text {
  margin: 0;
  color: #374151;
  font-size: clamp(1rem, 1.2vw, 1.125rem);
  line-height: 1.6;
}

/* Events Teaser (weißer Block) */
.events-teaser {
  background-color: #ffffff;
}
.events-teaser .section-title {
  color: #111827;
  font-size: clamp(1.875rem, 4vw, 2.5rem);
  font-weight: 600;
  margin-bottom: 1rem;
}
.events-grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(280px, 1fr));
  gap: 2rem;
  margin: 2rem 0;
}
.event-card {
  background-color: #ffffff;
  border-radius: 16px;
  padding: 1.75rem;
  box-shadow: 0 4px 16px rgba(31, 41, 55, 0.05);
  transition: transform 0.3s ease;
}
.event-card:hover {
  transform: translateY(-4px);
}
.event-date {
  font-size: 0.875rem;
  color: #6b7280;
  margin-bottom: 0.5rem;
  display: block;
}
.event-title {
  font-size: 1.25rem;
  font-weight: 600;
  margin-bottom: 0.75rem;
  color: #111827;
}
.event-excerpt {
  font-size: 1rem;
  color: #4b5563;
  line-height: 1.5;
}

/* Buttons */
.btn {
  display: inline-block;
  padding: 0.75rem 2rem;
  border-radius: 9999px;
  font-size: 1rem;
  font-weight: 500;
  text-decoration: none;
  transition: background-color 0.3s ease, transform 0.3s ease;
}
.btn-outline-gradient {
  border: 2px solid #0071e3;
  color: #0071e3;
  background: transparent;
}
.btn-outline-gradient:hover {
  background: #eff6ff;
  transform: translateY(-2px);
}

/* Scroll-triggered Animation */
.animate-on-scroll {
  opacity: 0;
  transform: translateY(20px);
  transition: all 0.6s ease;
}
.animate-in {
  opacity: 1 !important;
  transform: translateY(0) !important;
}

@media (max-width: 640px) {
  .stats-grid,
  .aktuales-grid,
  .events-grid {
    grid-template-columns: 1fr;
  }
  .btn {
    width: 100%;
    text-align: center;
  }
}
</style>
