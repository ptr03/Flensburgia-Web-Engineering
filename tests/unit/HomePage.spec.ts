// tests/unit/HomePage.spec.ts
import { shallowMount } from '@vue/test-utils'
import HomePage from '@/components/HomePage.vue'

describe('HomePage.vue', () => {
  beforeAll(() => {
    // stub IntersectionObserver globally for this suite
    // @ts-ignore
    window.IntersectionObserver = class {
      observe() {}
      unobserve() {}
      disconnect() {}
    }
  })

  it('renders the HeroSection stub', () => {
    const wrapper = shallowMount(HomePage, {
      global: {
        stubs: {
          'hero-section': true,
          'about-section': true,
        },
      },
    })
    expect(wrapper.find('hero-section-stub').exists()).toBe(true)
  })
})
