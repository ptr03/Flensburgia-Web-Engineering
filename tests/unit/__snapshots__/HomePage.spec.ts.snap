// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > renders without crashing 1`] = `
"<section data-v-9caf22ae="" data-v-42781596="" class="hero-section" style="background-image: url(&quot;/src/assets/pictures/3.jpeg&quot;);">
  <!-- Animated Title -->
  <div data-v-9caf22ae="" class="content-wrapper">
    <h1 data-v-9caf22ae="" class="hero-title">Schlaraffia Flensburgia</h1>
    <h2 data-v-9caf22ae="" class="welcome-text">Ein herzliches Willekum</h2>
  </div><!-- Button Grid overlapping next section -->
  <div data-v-9caf22ae="" class="button-grid">
    <router-link-stub data-v-9caf22ae="" to="/about" class="grid-button"></router-link-stub>
    <router-link-stub data-v-9caf22ae="" to="/sippungsfolge" class="grid-button"></router-link-stub>
    <router-link-stub data-v-9caf22ae="" to="/contact" class="grid-button"></router-link-stub>
  </div>
</section>
<!-- About Teaser -->
<section data-v-42781596="" class="section about-teaser">
  <div data-v-42781596="" class="container">
    <h2 data-v-42781596="" class="section-title text-center animate-on-scroll">Über Flensburgia</h2>
    <p data-v-42781596="" class="teaser-text animate-on-scroll"> Die Schlaraffia Flensburgia ist Teil der weltweiten Vereinigung Schlaraffia® – einer Gemeinschaft zur Pflege von Freundschaft, Kunst und Humor. </p>
    <div data-v-42781596="" class="stats-grid">
      <div data-v-42781596="" class="stat-item animate-on-scroll" style="transition-delay: 0s;"><svg data-v-42781596="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-calendar-days-icon lucide-calendar-days stat-icon text-primary">
          <path d="M8 2v4"></path>
          <path d="M16 2v4"></path>
          <rect width="18" height="18" x="3" y="4" rx="2"></rect>
          <path d="M3 10h18"></path>
          <path d="M8 14h.01"></path>
          <path d="M12 14h.01"></path>
          <path d="M16 14h.01"></path>
          <path d="M8 18h.01"></path>
          <path d="M12 18h.01"></path>
          <path d="M16 18h.01"></path>
        </svg>
        <h3 data-v-42781596="" class="stat-title">Jährliche Veranstaltungen</h3>
        <p data-v-42781596="" class="stat-value">30+</p>
      </div>
      <div data-v-42781596="" class="stat-item animate-on-scroll" style="transition-delay: 0.2s;"><svg data-v-42781596="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-trophy-icon lucide-trophy stat-icon text-primary">
          <path d="M6 9H4.5a2.5 2.5 0 0 1 0-5H6"></path>
          <path d="M18 9h1.5a2.5 2.5 0 0 0 0-5H18"></path>
          <path d="M4 22h16"></path>
          <path d="M10 14.66V17c0 .55-.47.98-.97 1.21C7.85 18.75 7 20.24 7 22"></path>
          <path d="M14 14.66V17c0 .55.47.98.97 1.21C16.15 18.75 17 20.24 17 22"></path>
          <path d="M18 2H6v7a6 6 0 0 0 12 0V2Z"></path>
        </svg>
        <h3 data-v-42781596="" class="stat-title">Gegründet</h3>
        <p data-v-42781596="" class="stat-value">1973</p>
      </div>
      <div data-v-42781596="" class="stat-item animate-on-scroll" style="transition-delay: 0.4s;"><svg data-v-42781596="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-users-icon lucide-users stat-icon text-primary">
          <path d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"></path>
          <path d="M16 3.128a4 4 0 0 1 0 7.744"></path>
          <path d="M22 21v-2a4 4 0 0 0-3-3.87"></path>
          <circle cx="9" cy="7" r="4"></circle>
        </svg>
        <h3 data-v-42781596="" class="stat-title">Aktive Mitglieder</h3>
        <p data-v-42781596="" class="stat-value">25+</p>
      </div>
    </div>
    <div data-v-42781596="" class="cta-wrapper text-center">
      <router-link-stub data-v-42781596="" to="/about" class="btn btn-outline-gradient mt-4 animate-on-scroll"></router-link-stub>
    </div>
  </div>
</section>
<!-- Aktuelles Section (in Blöcken, Hintergrund: weiß) -->
<section data-v-42781596="" class="section aktuelles-teaser">
  <div data-v-42781596="" class="container">
    <h2 data-v-42781596="" class="section-title text-center animate-on-scroll">Aktuelles</h2>
    <div data-v-42781596="" class="aktuales-grid">
      <div data-v-42781596="" class="aktual-card animate-on-scroll" style="transition-delay: 0.2s;">
        <p data-v-42781596="" class="aktual-text">Aktuelle Bilder der letzten Sippung in dieser Winterung.</p>
      </div>
      <div data-v-42781596="" class="aktual-card animate-on-scroll" style="transition-delay: 0.4s;">
        <p data-v-42781596="" class="aktual-text">Eine schöne und gute Sommerung!</p>
      </div>
      <div data-v-42781596="" class="aktual-card animate-on-scroll" style="transition-delay: 0.6000000000000001s;">
        <p data-v-42781596="" class="aktual-text">Sippungen jeden Donnerstag, Glocke 7:30 d. A. im Händelhof 7, 24394 Flensburg</p>
      </div>
    </div>
  </div>
</section>
<!-- Events Teaser (Kommende Veranstaltungen, verwaltet per JSON, Hintergrund: weiß) -->
<section data-v-42781596="" class="section events-teaser">
  <div data-v-42781596="" class="container">
    <h2 data-v-42781596="" class="section-title text-center animate-on-scroll">Kommende Veranstaltungen</h2>
    <div data-v-42781596="" class="events-grid">
      <article data-v-42781596="" class="event-card animate-on-scroll" style="transition-delay: 0.2s;"><time data-v-42781596="" class="event-date">2025-06-20</time>
        <h3 data-v-42781596="" class="event-title">Kaffeeklatsch bei den Erzschlaraffen</h3>
        <p data-v-42781596="" class="event-excerpt"></p>
      </article>
      <article data-v-42781596="" class="event-card animate-on-scroll" style="transition-delay: 0.4s;"><time data-v-42781596="" class="event-date">2025-07-05</time>
        <h3 data-v-42781596="" class="event-title">Liedersingen im Ehrenschlaraffenhof</h3>
        <p data-v-42781596="" class="event-excerpt"></p>
      </article>
      <article data-v-42781596="" class="event-card animate-on-scroll" style="transition-delay: 0.6000000000000001s;"><time data-v-42781596="" class="event-date">2025-08-10</time>
        <h3 data-v-42781596="" class="event-title">Sommerfest der Schlaraffen</h3>
        <p data-v-42781596="" class="event-excerpt"></p>
      </article>
    </div>
    <div data-v-42781596="" class="cta-wrapper text-center">
      <router-link-stub data-v-42781596="" to="/events" class="btn btn-outline-gradient mt-4 animate-on-scroll"></router-link-stub>
    </div>
  </div>
</section>"
`;

exports[`HomePage.vue > matches the snapshot 1`] = `
"<div data-v-42781596="" class="stub-hero"></div>
<!-- About Teaser -->
<section data-v-42781596="" class="section about-teaser">
  <div data-v-42781596="" class="container">
    <h2 data-v-42781596="" class="section-title text-center animate-on-scroll">Über Flensburgia</h2>
    <p data-v-42781596="" class="teaser-text animate-on-scroll"> Die Schlaraffia Flensburgia ist Teil der weltweiten Vereinigung Schlaraffia® – einer Gemeinschaft zur Pflege von Freundschaft, Kunst und Humor. </p>
    <div data-v-42781596="" class="stats-grid">
      <div data-v-42781596="" class="stat-item animate-on-scroll" style="transition-delay: 0s;"><svg data-v-42781596="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-calendar-days-icon lucide-calendar-days stat-icon text-primary">
          <path d="M8 2v4"></path>
          <path d="M16 2v4"></path>
          <rect width="18" height="18" x="3" y="4" rx="2"></rect>
          <path d="M3 10h18"></path>
          <path d="M8 14h.01"></path>
          <path d="M12 14h.01"></path>
          <path d="M16 14h.01"></path>
          <path d="M8 18h.01"></path>
          <path d="M12 18h.01"></path>
          <path d="M16 18h.01"></path>
        </svg>
        <h3 data-v-42781596="" class="stat-title">Jährliche Veranstaltungen</h3>
        <p data-v-42781596="" class="stat-value">30+</p>
      </div>
      <div data-v-42781596="" class="stat-item animate-on-scroll" style="transition-delay: 0.2s;"><svg data-v-42781596="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-trophy-icon lucide-trophy stat-icon text-primary">
          <path d="M6 9H4.5a2.5 2.5 0 0 1 0-5H6"></path>
          <path d="M18 9h1.5a2.5 2.5 0 0 0 0-5H18"></path>
          <path d="M4 22h16"></path>
          <path d="M10 14.66V17c0 .55-.47.98-.97 1.21C7.85 18.75 7 20.24 7 22"></path>
          <path d="M14 14.66V17c0 .55.47.98.97 1.21C16.15 18.75 17 20.24 17 22"></path>
          <path d="M18 2H6v7a6 6 0 0 0 12 0V2Z"></path>
        </svg>
        <h3 data-v-42781596="" class="stat-title">Gegründet</h3>
        <p data-v-42781596="" class="stat-value">1973</p>
      </div>
      <div data-v-42781596="" class="stat-item animate-on-scroll" style="transition-delay: 0.4s;"><svg data-v-42781596="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-users-icon lucide-users stat-icon text-primary">
          <path d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"></path>
          <path d="M16 3.128a4 4 0 0 1 0 7.744"></path>
          <path d="M22 21v-2a4 4 0 0 0-3-3.87"></path>
          <circle cx="9" cy="7" r="4"></circle>
        </svg>
        <h3 data-v-42781596="" class="stat-title">Aktive Mitglieder</h3>
        <p data-v-42781596="" class="stat-value">25+</p>
      </div>
    </div>
    <div data-v-42781596="" class="cta-wrapper text-center">
      <router-link-stub data-v-42781596="" to="/about" class="btn btn-outline-gradient mt-4 animate-on-scroll"></router-link-stub>
    </div>
  </div>
</section>
<!-- Aktuelles Section (in Blöcken, Hintergrund: weiß) -->
<section data-v-42781596="" class="section aktuelles-teaser">
  <div data-v-42781596="" class="container">
    <h2 data-v-42781596="" class="section-title text-center animate-on-scroll">Aktuelles</h2>
    <div data-v-42781596="" class="aktuales-grid">
      <div data-v-42781596="" class="aktual-card animate-on-scroll" style="transition-delay: 0.2s;">
        <p data-v-42781596="" class="aktual-text">Aktuelle Bilder der letzten Sippung in dieser Winterung.</p>
      </div>
      <div data-v-42781596="" class="aktual-card animate-on-scroll" style="transition-delay: 0.4s;">
        <p data-v-42781596="" class="aktual-text">Eine schöne und gute Sommerung!</p>
      </div>
      <div data-v-42781596="" class="aktual-card animate-on-scroll" style="transition-delay: 0.6000000000000001s;">
        <p data-v-42781596="" class="aktual-text">Sippungen jeden Donnerstag, Glocke 7:30 d. A. im Händelhof 7, 24394 Flensburg</p>
      </div>
    </div>
  </div>
</section>
<!-- Events Teaser (Kommende Veranstaltungen, verwaltet per JSON, Hintergrund: weiß) -->
<section data-v-42781596="" class="section events-teaser">
  <div data-v-42781596="" class="container">
    <h2 data-v-42781596="" class="section-title text-center animate-on-scroll">Kommende Veranstaltungen</h2>
    <div data-v-42781596="" class="events-grid">
      <article data-v-42781596="" class="event-card animate-on-scroll" style="transition-delay: 0.2s;"><time data-v-42781596="" class="event-date">2025-06-20</time>
        <h3 data-v-42781596="" class="event-title">Kaffeeklatsch bei den Erzschlaraffen</h3>
        <p data-v-42781596="" class="event-excerpt"></p>
      </article>
      <article data-v-42781596="" class="event-card animate-on-scroll" style="transition-delay: 0.4s;"><time data-v-42781596="" class="event-date">2025-07-05</time>
        <h3 data-v-42781596="" class="event-title">Liedersingen im Ehrenschlaraffenhof</h3>
        <p data-v-42781596="" class="event-excerpt"></p>
      </article>
      <article data-v-42781596="" class="event-card animate-on-scroll" style="transition-delay: 0.6000000000000001s;"><time data-v-42781596="" class="event-date">2025-08-10</time>
        <h3 data-v-42781596="" class="event-title">Sommerfest der Schlaraffen</h3>
        <p data-v-42781596="" class="event-excerpt"></p>
      </article>
    </div>
    <div data-v-42781596="" class="cta-wrapper text-center">
      <router-link-stub data-v-42781596="" to="/events" class="btn btn-outline-gradient mt-4 animate-on-scroll"></router-link-stub>
    </div>
  </div>
</section>"
`;
